src/main/java/org/springframework/integration/kafka/dsl/Kafka.java

```diff
@@ -74,72 +76,73 @@
 	 * Create an initial {@link KafkaInboundChannelAdapterSpec} with the consumer factory and
 	 * topics.
 	 * @param consumerFactory the consumer factory.
-	 * @param topics the topic(s).
+	 * @param consumerProperties the consumerProperties.
 	 * @param <K> the Kafka message key type.
 	 * @param <V> the Kafka message value type.
 	 * @return the spec.
 	 * @since 3.0.1
 	 */
 	public static <K, V> KafkaInboundChannelAdapterSpec<K, V> inboundChannelAdapter(
-			ConsumerFactory<K, V> consumerFactory, String... topics) {
+			ConsumerFactory<K, V> consumerFactory, ConsumerProperties consumerProperties) {
```

Comment:
This is a breaking change. We could allow to do such a radical modification only in the next major version.
Right now we only have a choice to keep old ctos as `@Deprecated` and we may remove them in the next minor version then.

I know it is hard work to do in this code base, but we don't have a choice to provide a backward compatible code with a smooth migration option.

Thanks for udertstanding.

Reply:
Okay, I am guessing we'd also want to maintain backwards compatibility with XSD syntax then?

Reply:
That’s correct assumption. With mentioning in the old attribute descriptions that they are deprecated and will be removed in the next version. Thank you again!

Reply:
Okay, I'll have an update soon. Is it okay to remove the required flag for `group-id` though?
Then in `KafkaInboundChannelAdapterParser` we can check if `consumer-factory` is set then use that otherwise use raw properties. 

https://github.com/spring-projects/spring-integration-kafka/blob/1903849cea7bf035a373309868d0fc8a2252f100/src/main/resources/org/springframework/integration/kafka/config/spring-integration-kafka-3.2.xsd#L179-L185

Reply:
Yes. It doesn’t look like breaking change to me

Reply:
Restored all pre-3.2 constructors and setters. Added depreciation warning to them.